gems/pending/disk/modules/miq_disk_cache.rb

```diff
@@ -0,0 +1,113 @@
+require 'rufus/lru'
+require_relative "../MiqDisk"
+require 'ostruct'
+
+module MiqDiskCache
+  MIN_SECTORS_TO_CACHE = 32
+  DEF_CACHE_SIZE       = 100
+  DEBUG_CACHE_STATS    = false
+
+  attr_reader :d_size, :blockSize, :cache_size, :sectors_to_cache, :cache_hits, :cache_misses
+
+  def self.new(down_stream, d_info = nil)
```

Comment:
Maybe, instead of passing in a `dInfo` we should pass in `cache_size` and `sectors_to_cache` explicitly. For now, we'll still have to stuff them into a `dInfo` for the `MiqDisk.new` call, but when we make this a proper class, it'll be cleaner.